### PR TITLE
Refactoring: Change provenance to composition, Splitting up discovery.go, and Cleaning up

### PR DESCRIPTION
**Pull request by djarotech**
Changes to types.go and discovery.go

-Renaming variables and splitting up the coe into multiple files
- Also, my atom editor is uses gofmt on the code after every save, so it did clean up some small things like extra spaces, spaces, inconsistent spacing, import organizing, etc. 

Fixes Issues: https://github.com/cloud-ark/kubediscovery/issues/28 and https://github.com/cloud-ark/kubediscovery/issues/37

### CODE REVIEW

**Comment by djarotech** [comment]
https://github.com/cloud-ark/kubediscovery/issues/37

**Comment by devdattakulkarni** [comment]
Please try the steps on https://github.com/cloud-ark/kubeplus/pull/211 to build a new kubediscovery image and check if it works.

**Comment by djarotech** [comment]
Worked !

File: pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -202,23 +202,23 @@ func handleComposition(request *restful.Request, response *restful.Response) {
...
+			firstPart = strings.ToUpper(firstPart)
+			secondPart := strings.Join(parts[1:], "")
+			resourceKind = firstPart + secondPart
 	*/
 
-	fmt.Printf("Kind:%s, Instance:%s\n", resourceKind, resourceInstance) 
+	fmt.Printf("Kind:%s, Instance:%s\n", resourceKind, resourceInstance)
 
-	describeInfo := discovery.TotalClusterProvenance.GetProvenance(resourceKind, resourceInstance)
+	describeInfo := discovery.TotalClusterCompositions.GetProvenance(resourceKind, resourceInstance)
```
Comment (devdattakulkarni) [inline pkg/apiserver/apiserver.go]: GetProvenance -> GetCompositions

File: pkg/apiserver/apiserver.go [hunk 2]
```diff
@@ -255,40 +255,40 @@ func getCompositions(request *restful.Request, response *restful.Response) {
...
-	//provenance.TotalClusterProvenance.PrintProvenance()
+	//provenance.TotalClusterCompositions.PrintCompositions()
 
 	// Path looks as follows:
 	// /apis/kubediscovery.cloudark.io/v1/namespaces/default/deployments/dep1/compositions
 	resourcePathSlice := strings.Split(requestPath, "/")
 	resourceKind := resourcePathSlice[6] // Kind is 7th element in the slice
 	fmt.Printf("Resource Kind:%s, Resource name:%s\n", resourceKind, resourceName)
-	provenanceInfo := discovery.TotalClusterProvenance.GetProvenance(resourceKind, resourceName)
+	provenanceInfo := discovery.TotalClusterCompositions.GetProvenance(resourceKind, resourceName)
```
Comment (devdattakulkarni) [inline pkg/apiserver/apiserver.go]: GetProvenance -> GetCompositions

File: pkg/discovery/discovery.go [hunk 3]
```diff
@@ -140,18 +135,18 @@ func BuildCompositionTree() {
...
 }
 
-func (cp *ClusterProvenance) checkIfProvenanceNeeded(resourceKind, resourceName string) bool {
+func (cp *ClusterCompositions) checkIfProvenanceNeeded(resourceKind, resourceName string) bool {
 	cp.mux.Lock()
 	defer cp.mux.Unlock()
-	for _, provenanceItem := range cp.clusterProvenance {
-		kind := provenanceItem.Kind
-		name := provenanceItem.Name
+	for _, compItem := range cp.clusterCompositions {
```
Comment (devdattakulkarni) [inline pkg/discovery/discovery.go]: compItem -> compositionItem

File: pkg/discovery/discovery.go [hunk 4]
```diff
@@ -489,117 +310,67 @@ func (cp *ClusterProvenance) GetProvenance(resourceKind, resourceName string) st
...
+				presentList = append(presentList, comp)
 			}
 		}
 	}
 	//fmt.Printf("Updated Cluster Prov List:%v\n", presentList)
-	cp.clusterProvenance = presentList
+	cp.clusterCompositions = presentList
 }
 
 // This stores Provenance information in memory. The provenance information will be lost
```
Comment (devdattakulkarni) [inline pkg/discovery/discovery.go]: Provenance Information -> Composition Information

File: pkg/discovery/openapispec.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
+package discovery
```
Comment (devdattakulkarni) [inline pkg/discovery/openapispec.go]: This file will serve other kinds of documentation as well besides OpenAPISpec for an Operator

So lets rename it to: operatordocs.go

File: pkg/discovery/types.go [hunk 6]
```diff
@@ -6,56 +6,56 @@ import (
...
-	Children []MetaDataAndOwnerReferences
+	Children  []MetaDataAndOwnerReferences
 }
 
 // Used for intermediate storage -- probably can be merged with Composition
-type Provenance struct {
-	Kind string
-	Name string
-	Status string
+type Compositions struct {
```
Comment (devdattakulkarni) [inline pkg/discovery/types.go:42]: Lets keep this singular - Composition

File: pkg/discovery/types.go [hunk 7]
```diff
@@ -6,56 +6,56 @@ import (
...
+	Status          string
 	CompositionTree *[]CompositionTreeNode
 }
 
 // Used to hold entire composition Provenance of all the Kinds
-type ClusterProvenance struct {
-	clusterProvenance []Provenance
-	mux sync.Mutex
+type ClusterCompositions struct {
+	clusterCompositions []Compositions
```
Comment (devdattakulkarni) [inline pkg/discovery/types.go:51]: Once above change is made, this will be []Composition

**Comment by devdattakulkarni** [comment]
I have added comments. Once you make the changes, try it in your set by following the steps in:

https://github.com/cloud-ark/kubeplus/blob/master/docs/setup/local-testing.txt

File: pkg/discovery/types.go (hunk 6 above)
Comment (djarotech) [reply pkg/discovery/types.go:42]: There already exists a type Composition

Comment (djarotech) [reply pkg/discovery/types.go:42]: Will create another issue for this

File: pkg/discovery/types.go (hunk 7 above)
Comment (djarotech) [reply pkg/discovery/types.go:51]: Will create another issue for this

File: pkg/apiserver/apiserver.go (hunk 1 above)
Comment (devdattakulkarni) [reply pkg/apiserver/apiserver.go]: Can you change GetProvenance function to GetCompositions

File: pkg/discovery/types.go (hunk 6 above)
Comment (devdattakulkarni) [reply pkg/discovery/types.go:42]: +1

**Pull request merged**